src/modules/launcher/PowerLauncher/MainWindow.xaml.cs

```diff
@@ -98,32 +100,16 @@ private void SuggestionsList_PreviewMouseLeftButtonUp(object sender, MouseButton
 
         private void ViewModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
         {
-            if (e.PropertyName == nameof(MainViewModel.MainWindowVisibility))
+            if (Visibility == System.Windows.Visibility.Visible)
             {
-                if (Visibility == System.Windows.Visibility.Visible)
-                {
-                    _deletePressed = false;
-                    _firstDeleteTimer.Start();
-                    Activate();
-                    //(this.FindResource("IntroStoryboard") as Storyboard).Begin();
-
-                    UpdatePosition();
-                    SearchBox.QueryTextBox.Focus();
-                    _settings.ActivateTimes++;
-
-                    if (!_viewModel.LastQuerySelected)
-                    {
-                        _viewModel.LastQuerySelected = true;
-                    }
+                // Not called on first launch
+                // Additionally called when deactivated by clicking on screen  
+                UpdatePosition();

```

Comment:
Can you elaborate on why having this in the function window_isVisibleChanged() is causing an issue?

Reply:
Calling `updatePosition` in the `Window_IsVisibleChanged` callback results in incorrect computation of Left Coordinate for Main window. This is because `MainWindow.ActualWidth` hasn't been calculated in the WPF lifecycle yet. Hence, `WindowsLeft` function will give incorrect value of Left coordinate.